cpu/sam0_common/periph/gpio.c

```diff
@@ -217,15 +232,55 @@ int gpio_init_int(gpio_t pin, gpio_mode_t mode, gpio_flank_t flank,
     _EIC->WAKEUP.reg |= (1 << exti);
     /* enable the EIC module*/
     _EIC->CTRL.reg = EIC_CTRL_ENABLE;
-    while (_EIC->STATUS.reg & EIC_STATUS_SYNCBUSY) {}
+    EIC_SYNC();
 #else /* CPU_FAM_SAML21 */
     /* enable the EIC module*/
     _EIC->CTRLA.reg = EIC_CTRLA_ENABLE;
-    while (_EIC->SYNCBUSY.reg & EIC_SYNCBUSY_ENABLE) {}
+    EIC_SYNC();
 #endif
     return 0;
 }
 
+inline static void reenable_eic(gpio_eic_clock_t clock) {
+#if !defined(CPU_SAMD21)
+    uint32_t ctrla_reg = EIC_CTRLA_ENABLE;
+
+    EIC->CTRLA.reg = 0;
+    EIC_SYNC();
+
+    if (clock == _EIC_CLOCK_SLOW) {
+        ctrla_reg |= EIC_CTRLA_CKSEL;
+    }
+
+    EIC->CTRLA.reg = ctrla_reg;
+    EIC_SYNC();
+#endif
+}
+
+void gpio_pm_cb_enter(int deep)
+{
+#if defined(PM_SLEEPCFG_SLEEPMODE_STANDBY)
+    (void) deep;
```

Comment:
Why introduce a `deep` parameter when you are not using it?
Checking the `SLEEPMODE` is more robust anyway.

Reply:
I noticed that the `samd21` uses deep to distinguish between STANDBY and IDLE modes. If we want to detect this mode on future callbacks, we need to know the value of `deep`.

But we could add this later, if we actually need it.

Reply:
Ah good catch!
But tbh, this is not a public API, so we can extend it any time when adding support for `samd2x`.

Reply:
I mean it matches the signature of `cortexm_sleep()` and it makes implementing support for samd2x easier.
I'm fine with keeping it too - I guess I just never liked that `deep` variable because it's so opaque.

Reply:
Tried it on `samr21-xpro`, works like a charm 😄